src/Source/SourceIteratorInterface.php

```diff
@@ -14,8 +14,7 @@
 namespace Sonata\Exporter\Source;
 
 /**
- * @phpstan-template T of array
- * @phpstan-extends \Iterator<T>
+ * @phpstan-extends \Iterator<array<mixed>>
```

Comment:
Couldn't we use the template for the array elements?
```
@phpstan-template T
@phpstan-extends \Iterator<array<array-key, T>>
```

Reply:
If you return `array{0: int, 1: string}`, you'll have to use `T = int|string`.

The day support for `array-shape` in template will be supported, we'll have to make a BC-break change from `T = int|string` to `T = array<int|string>` (or `T = array{0: int, 1: string}`) ; so I'm not sure about the benefit.

The other option was to write
```
@phpstan-template T
@phpstan-extends \Iterator<T>
```
and not bound T to array.

But using `T = array<>` or `T = array{}` is not well supported, so it will still be kinda useless.

IMHO silencing this generic could be a simple solution for now. I'm pretty sure this generic will (almost) never be used ; for instance in SonataAdmin, we used `SourceIteratorInterface<array<mixed>>`.